src/components/home/AllCollections.js

```diff
@@ -109,31 +119,23 @@ function AllCollections() {
     },
   );
 
-  // get all categories
-  const { data: allCategories } = hooks.useCategories();
-
-  // get category map (to map between category id and name)
-  const categoriesMap = new Map(
-    allCategories?.map((entry) => [entry.name, entry.id]),
+  // get categories in each type
+  const { data: categoryTypes } = hooks.useCategoryTypes();
+  const { data: categories } = hooks.useCategories();
+  const allCategories = categories?.groupBy((entry) => entry.type);
+  const levelList = allCategories?.get(
+    categoryTypes?.find((type) => type.name === LEVEL)?.id,
   );
+  const disciplineList = allCategories
+    ?.get(categoryTypes?.find((type) => type.name === DISCIPLINE)?.id)
+    ?.sort(compare);
 
-  // get collections for given category
-  const getCollections = (discipline) => {
-    const { data } = hooks.useItemsInCategories([
-      categoriesMap?.get(discipline),
-    ]);
-    return data;
-  };
-
-  // get collections
-  const collectionsMath = getCollections(MATH);
-  const collectionsLiterature = getCollections(LITERATURE);
-  const collectionsNaturalScience = getCollections(NATURAL_SCIENCE);
-  const collectionsSocialScience = getCollections(SOCIAL_SCIENCE);
-  const collectionsLanguage = getCollections(LANGUAGE);
-  const collectionsArt = getCollections(ART);
+  // state variable to record selected options
+  const [selectedLevels, setSelectedLevels] = useState(false);
+  const [selectedDisciplines, setSelectedDisciplines] = useState(false);
```

Comment:
Is there actually a reason for splitting the selection? I think it can also work without splitting it.

Reply:
I merge them into one. However I do need somehow distinguish between if it's level or discipline, so if user choose a different one, it can switch to the new category. This problem will be completely solved after multiselecting is implemented I think.